src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs

```diff
@@ -26,32 +27,35 @@ internal protected override void Open()
 			}
 		}
 
-		internal protected override void Close()
+		private void SaveInitialTime() => _initialTime = Time;
+
+		internal protected override void Close() { base.Close(); }
+
+		private void SaveTime(int hourOfDay, int minutes)
 		{
-			_dialog?.Dismiss();
+			if (Time.Hours != hourOfDay || Time.Minutes != minutes)
+			{
+				Time = new TimeSpan(Time.Days, hourOfDay, minutes, Time.Seconds, Time.Milliseconds);
+				SaveInitialTime();
+			}
 		}
 
 		private bool UseAlertDialogWorkAround()
 		{
-			// WARNING Adding any case that uses API lower than 6.0 will require the use of deprecated properties 
-			// on the TimePicker i.e. CurrentHour and CurrentMinute, server build rules may need to be modified or
-			// a #prama directive may be needed.
-
-			var ver = Android.OS.Build.VERSION.Release;
 			//On Samsung devices >= 6.0 the TimePickerDialog is not displayed properly in Landscape.
-			return Android.OS.Build.VERSION.Release.StartsWith("6.") && //Any 6.x.x version				
-				   Android.OS.Build.Manufacturer.IndexOf("samsung", StringComparison.OrdinalIgnoreCase) >= 0;
+			return Build.VERSION.Release.StartsWith("6.") && //Any 6.x.x version				
+				   Build.Manufacturer.Contains("samsung", StringComparison.OrdinalIgnoreCase);
```

Comment:
This is a merge issue. This will not build using the latest master due to 15.8 compatibility restrictions

Reply:
so i have to keep Android.OS.?